server/app/services/mongodb/migrator.rb

```diff
@@ -8,7 +8,7 @@ class Migrator
     include DistributedLocks
 
     LOCK_NAME = 'mongodb_migrate'.freeze
-    LOCK_TIMEOUT = 60
+    LOCK_TIMEOUT = (60 * 5)
```

Comment:
I'm thinking we should not have any timeout for the migration lock, i.e. fix `Migrator#migrate` to use `with_dlock(LOCK_NAME, 0)`? Then the puma worker timeout will kill it, and allow it to re-aquire the lock after cleaning up any stale locks.